go/arrow/compute/exprs/types.go

```diff
@@ -623,7 +623,7 @@ func ToSubstraitType(dt arrow.DataType, nullable bool, ext ExtensionIDSet) (type
 		if err != nil {
 			return nil, err
 		}
-		valueType, err := ToSubstraitType(dt.ValueType(), dt.ValueField().Nullable, ext)
+		valueType, err := ToSubstraitType(dt.Elem(), dt.ElemField().Nullable, ext)
```

Comment:
This is wrong, as the value here is a value in `key-value pair` sense, so it has to be `Item` in `arrow.MapType`